expression/builtin_time.go

```diff
@@ -1994,7 +1994,11 @@ func (b *builtinSysDateWithFspSig) evalTime(row chunk.Row) (d types.Time, isNull
 	}
 
 	loc := b.ctx.GetSessionVars().Location()
-	now := time.Now().In(loc)
+	nowTs, err := getStmtTimestamp(b.ctx)
+	if err != nil {
+		return types.Time{}, true, err
+	}
+	now := nowTs.In(loc)
```

Comment:
Why change the implement of `builtinSysDateWithFspSig`?  `getStmtTimestamp` will return the cached time and `time.Now()` will return current local time. And see the document , I think we should use `time.Now()`. Or I am misunderstanding?

Reply:
If use `time.Now()` then `expressions` tests will always fail:

```sh
builtin_time_vec_test.go:223:
    testVectorizedBuiltinFunc(c, vecBuiltinTimeCases)
bench_test.go:989:
    c.Assert(val, Equals, t64s[i], commentf(i))
... obtained types.Time = types.Time{Time:types.MysqlTime{hour:0x11, microsecond:0x1b9cc, year:0x7e3, month:0xb, day:0xb, minute:0xf, second:0x1a}, Type:0xc, Fsp:4} ("2019-11-11 17:15:26.1131")
... expected types.Time = types.Time{Time:types.MysqlTime{hour:0x11, microsecond:0x1b968, year:0x7e3, month:0xb, day:0xb, minute:0xf, second:0x1a}, Type:0xc, Fsp:4} ("2019-11-11 17:15:26.1130")
```

The reason of the fail: the difference between in fsp for two consecutive call of the `time.Now()`. 

That's why original `builtinSysDateWithFspSig` was changed to use `getStmtTimestamp` instead of `time.Now()`.

Reply:
I think it is a bug of test framework.  @qw4990  PTAL

Reply:
Unlike `Now`, which returns a constant time that indicates the time at which the statement began to execute, `SysDate` returns the exact time at which it executes. So like `Rand`, `SysDate` is nondeterministic and we can't use our testing framework to test it.
So please write a new test function to test it, for example:
```
begin := time.Now()
sysDate.VecEval(result)
end := time.Now()
for time := range result {
    assert(time between begin to end)
}
```

Reply:
Looks like we have a solution - #13432 
Will fix when #13425 will be merged.

Reply:
Fixed.